packages/web/src/features/translation/TranslationSidebar.tsx

```diff
@@ -29,295 +36,315 @@ type TranslationSidebarProps = {
-  const bdbCurrentVerseRef = `${
-    bdbBookRefNames[bookId - 1]
-  } ${chapterNumber}:${verseNumber}`;
+    const lexiconEntryRef = useRef<HTMLDivElement>(null);
+    const [previewElement, setPreviewElement] = useState<HTMLDivElement | null>(
+      null
+    );
+    const [previewVerseIds, setPreviewVerseIds] = useState<string[]>([]);
 
-  const lexiconEntryRef = useRef<HTMLDivElement>(null);
-  const [previewElement, setPreviewElement] = useState<HTMLDivElement | null>(
-    null
-  );
-  const [previewVerseIds, setPreviewVerseIds] = useState<string[]>([]);
+    useEffect(() => {
+      const { current } = lexiconEntryRef;
+      // Highlight references to the currently selected verse
+      current
+        ?.querySelectorAll(`a[data-ref="${bdbCurrentVerseRef}"]`)
+        .forEach((element) => element.classList.add('bg-yellow-300'));
+    }, [bdbCurrentVerseRef, lexiconEntry, t]);
 
-  useEffect(() => {
-    const { current } = lexiconEntryRef;
-    // Highlight references to the currently selected verse
-    current
-      ?.querySelectorAll(`a[data-ref="${bdbCurrentVerseRef}"]`)
-      .forEach((element) => element.classList.add('bg-yellow-300'));
-  }, [bdbCurrentVerseRef, lexiconEntry, t]);
+    const openPreview = (anchorElement: HTMLAnchorElement) => {
+      const oldPreview = document.querySelector('#ref-preview');
+      oldPreview?.remove();
 
-  const openPreview = (anchorElement: HTMLAnchorElement) => {
-    const oldPreview = document.querySelector('#ref-preview');
-    oldPreview?.remove();
+      const reference = anchorElement.getAttribute('data-ref') ?? '';
+      setPreviewVerseIds(parseReferenceRange(reference, t));
 
-    const reference = anchorElement.getAttribute('data-ref') ?? '';
-    setPreviewVerseIds(parseReferenceRange(reference, t));
+      const previewElement = document.createElement('div');
+      previewElement.id = 'ref-preview';
+      previewElement.style.width = 'calc(100% + 32px)';
+      previewElement.style.margin = '4px -16px';
+      previewElement.style.padding = '8px 16px';
+      previewElement.style.backgroundColor = '#ffffff80';
+      previewElement.style.float = 'left';
+      anchorElement.insertAdjacentElement('afterend', previewElement);
+      setPreviewElement(previewElement);
+    };
 
-    const previewElement = document.createElement('div');
-    previewElement.id = 'ref-preview';
-    previewElement.style.width = 'calc(100% + 32px)';
-    previewElement.style.margin = '4px -16px';
-    previewElement.style.padding = '8px 16px';
-    previewElement.style.backgroundColor = '#ffffff80';
-    previewElement.style.float = 'left';
-    anchorElement.insertAdjacentElement('afterend', previewElement);
-    setPreviewElement(previewElement);
-  };
+    const [tabIndex, setTabIndex] = useState(0);
 
-  return (
-    <div
-      className={`
+    useImperativeHandle(ref, () => ({
+      openNotes: () => {
+        flushSync(() => setTabIndex(1));
+        translatorNotesRef.current?.focus();
```

Comment:
The focus isn't working unless I press the button again after the tab is correct. What is the `flushSync` doing?

Reply:
Oops, I forgot to push my changes. Sorry about that. Done.

Reply:
It works great now! 👍🏽